net/server.h

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+// C++ includes
+#include <memory>
+#include <string>
+#include <vector>
+
+// OpenCV
+#include <opencv2/opencv.hpp>
+
+// GeNN robotics includes
+#include "../robots/motor.h"
+#include "../video/input.h"
+
+// local includes
+#include "node.h"
+#include "socket.h"
+
+namespace GeNNRobotics {
+namespace Net {
+class Server : public Node
+{
+public:
+    Server(int port = Socket::DefaultListenPort);
+    virtual ~Server();
+    Socket *getSocket() const override;
+    void run() override;
+
+protected:
+
+private:
+    socket_t m_ListenSocket = INVALID_SOCKET;
+    std::unique_ptr<Socket> m_Socket;
+};
+
+/*
+ * Create a server to send motor commands
+ */
+Server::Server(int port)
+{
+    struct sockaddr_in addr;
+    int on = 1;
+
+    m_ListenSocket = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+    if (m_ListenSocket == INVALID_SOCKET) {
+        goto error;
+    }
+
+#ifndef _WIN32
+    if (setsockopt(m_ListenSocket, SOL_SOCKET, SO_REUSEADDR, &on, sizeof(on)) <
+        0) {
+        goto error;
+    }
+#endif
+
+    memset(&addr, '0', sizeof(addr));
```

Comment:
I'm pretty sure you don't want to do this - this filles the struct with the ASCII  code for 0 :smile:

Reply:
Haha. Oops.